src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -855,6 +860,174 @@ public uint256 LastReceivedBlockHash()
             return lastBlockSyncedHash;
         }
 
+        /// <summary>
+        /// Please see https://github.com/bitcoin/bitcoin/blob/726d0668ff780acb59ab0200359488ce700f6ae6/src/wallet/wallet.cpp#L3641
+        /// </summary>
+        /// <returns>A grouped list of base 58 addresses.</returns>
+        public List<List<string>> GetAddressGroupings()
+        {
+            // Get the wallet to check.
+            var walletName = this.GetDefaultOrFirstWalletName();
```

Comment:
Should we be passing the walelt name in to the method? (as well as the account)

Reply:
The difficulty with that is that the Core RPCs were originally designed to operate on a single wallet, and thus there isn't really a mechanism to specify which wallet you want (and to authenticate against it for RPCs that sign, for example).

Reply:
Yes but in that case the selection of wallet should happen at controller level I would think (@mikedennis had similar issues when preparing for NBitcoin)

Reply:
My point is such logic of wallet default name should be considered outside of the WalletManager

Reply:
@dangershony part of `GetDefaultOrFirstWalletName` logic was in the WalletRPCController but because we now need this in the WalletManager I moved it here. I supposed I can move it back but then it duplicates the logic.

Reply:
Typically WalletManager takes in the wallet/account name as parameters as the Api typically passes those in. RPC is different as bitcoin doesn't have the concept of multiple wallets. So yes I would do the "GetDefaultOrFirstWalletName in the rpc controller (just like the other wallet methods there) and pass in the wallet/account name  into the walletmanager method.

Reply:
OK yeah good suggestion @mikedennis :) I will make that change.